File: Extensions/Tessa.Extensions.Default.Client/AbTest/AbExternalFileCreationToken.cs
#nullable enable
using Tessa.Files;

namespace Tessa.Extensions.Default.Client.AbTest
{
    public class AbExternalFileCreationToken : FileCreationToken
    {
        #region Properties

        /// <summary>
        /// Описание файла.
        /// </summary>
        public string? Description { get; set; }

        #endregion

        #region Base Overrides

        /// <inheritdoc/>
        protected override void SetCore(IFileCreationToken token)
        {
            base.SetCore(token);
            this.Description = (token as AbExternalFileCreationToken)?.Description;
        }

        /// <inheritdoc/>
        protected override void SetCore(IFile file)
        {
            base.SetCore(file);
            this.Description = (file as AbExternalFile)?.Description;
        }

        #endregion
    }
}

File: WebClient SDK/src/default/abTest/abExternalFileCreationToken.ts
import { AbExternalFile } from './abExternalFile';
import { FileCreationToken, IFile } from 'tessa/files';

export class AbExternalFileCreationToken extends FileCreationToken {
  public description: string | Blob;

  public set(file: IFile): void {
    super.set(file);

    this.description = file instanceof AbExternalFile ? file.description : '';
  }
}
